linter/linter.go

```diff
@@ -15,45 +17,82 @@ var StatusIntMap = map[int]string{
 }
 
 type LintStruct struct {
-	info   string                          // description of the lint
-	source string                          // source of the lint
-	exec   func(resp *ocsp.Response) error // the linting function itself
+	Info   string                          // description of the lint
+	Source string                          // source of the lint
+	Exec   func(resp *ocsp.Response, leafCert *x509.Certificate) (LintStatus, string) // the linting function itself
 }
 
 // Lints is the global array of lints that are to be tested (TODO: change to a map)
-var Lints = []LintStruct{
-	{
-		"Check that response producedAt date is no more than " + ProducedAtLimit + " in the past",
-		"Apple Lint 03",
+var Lints = []*LintStruct{
+	&LintStruct{
+		"Check response producedAt date",
+		"Apple Lints 03 & 05",
 		LintProducedAtDate,
 	},
-	{
-		"Check that response thisUpdate date is no more than " + ThisUpdateLimit + " in the past",
-		"Apple Lint 03",
+	&LintStruct{
+		"Check response thisUpdate date",
+		"Apple Lints 03 & 05",
 		LintThisUpdateDate,
 	},
 }
 
+type LintStatus string
+
+const (
+	Passed LintStatus = "PASSED" // lint passed
+	Failed LintStatus = "FAILED" // lint failed
+	Error LintStatus = "ERROR" // encountered error while running lint
+)
+
+type LintResult struct {
+	Lint *LintStruct
+	Status LintStatus
+	Info string
+}
+
 type LinterInterface interface {
-	LintOCSPResp(*ocsp.Response)
+	LintOCSPResp(*ocsp.Response, *x509.Certificate, bool)
 }
 
 type Linter struct{}
 
-// LintOCSPResp takes in a parsed OCSP response and prints its status
-// TODO: change function so that it returns a list of failed lints
-func (l Linter) LintOCSPResp(resp *ocsp.Response) {
-	log.Println("OCSP Response status: " + StatusIntMap[resp.Status]) // placeholder
+// printResults prints the results of all the lints run
+func printResults(results []*LintResult, verbose bool) {
+	fmt.Println("Printing lint results: ")
+	// sort by status so printing prints all the lints that errored, then failed, then passed
+	sort.Slice(results, func(i, j int) bool {
+		return results[i].Status < results[j].Status
+	})
 
-	log.Println("Linting OCSP Response...")
-	for _, lint := range Lints {
-		log.Print(lint.info + ": ")
-		err := lint.exec(resp)
-		if err == nil {
-			log.Println("passed")
-		} else {
-			log.Println("failed: " + err.Error())
+	allPassed := true
+
+	for _, result := range results {
+		if result.Status != Passed {
+			allPassed = false
```

Comment:
I think you can also do something like `allPassed &= (result.Status != Passed)`

Reply:
I don't think Go has such a syntax unfortunately